build: bpftool fails to build with clang as a default compiler

Attempt to build the project on a system with clang as a main compiler (in my case it's Gentoo with musl-llvm profile, where there is no GCC):

```
$ cc --version
clang version 19.1.4+libcxx
Target: x86_64-pc-linux-musl
Thread model: posix
InstalledDir: /usr/lib/llvm/19/bin
Configuration file: /etc/clang/x86_64-pc-linux-musl-clang.cfg
```

fails with:

```
[7/51] Generating bpftool_target with a custom command
FAILED: bpftool_dummy.c.__PHONY__
/bin/bash /home/vadorovsky/src/scx/meson-scripts/build_bpftool /usr/bin/jq /usr/bin/make /home/vadorovsky/src/scx/build/bpftool/src 32
make[1]: gcc: No such file or directory
make[1]: gcc: No such file or directory
make[1]: *** [Makefile:134: /home/vadorovsky/src/scx/build/bpftool/src/bootstrap/libbpf/staticobjs/bpf.o] Error 127
make[1]: *** Waiting for unfinished jobs....
make[1]: gcc: No such file or directory
make[1]: *** [Makefile:134: /home/vadorovsky/src/scx/build/bpftool/src/bootstrap/libbpf/staticobjs/btf.o] Error 127
make[1]: gcc: No such file or directory
make[1]: *** [Makefile:134: /home/vadorovsky/src/scx/build/bpftool/src/bootstrap/libbpf/staticobjs/libbpf.o] Error 127
make[1]: gcc: No such file or directory
make[1]: *** [Makefile:134: /home/vadorovsky/src/scx/build/bpftool/src/bootstrap/libbpf/staticobjs/libbpf_errno.o] Error 127
make: *** [Makefile:52: /home/vadorovsky/src/scx/build/bpftool/src/bootstrap/libbpf/libbpf.a] Error 2
make: *** Waiting for unfinished jobs....
```

I usually can build bpftool (and the entire Linux monorepo) with clang using `make LLVM=1`. I think the best workaround would be passing `LLVM=1` in `meson-scripts/build_bpftool` if the compiler is clang.

--- a/meson-scripts/build_bpftool
+++ b/meson-scripts/build_bpftool
@@ -23,6 +23,13 @@ for arg in ${args[@]:(idx+1)}; do
     esac
 done
 
+extra_args=()
+# When using clang, `LLVM=1` needs to be passed as an argument. Otherwise some
+# parts of the build system will attempt to invoke GCC.
+if cc --version 2>&1 | grep clang; then
+    extra_args+="LLVM=1"
+fi
+
 cd $3
-make_out=$(env CC="$cc" CFLAGS="$cflags" "$2" -j"$4")
+make_out=$(env CC="$cc" CFLAGS="$cflags" "$2" -j"$4" $extra_args)
 exit $?

